lib/matplotlib/backend_bases.py

```diff
@@ -1842,11 +1842,10 @@ def inaxes(self, xy):
         Returns
         -------
         axes : `~matplotlib.axes.Axes` or None
-            The topmost axes containing the point, or None if no axes.
+            The topmost visible axes containing the point, or None if no axes.
         """
         axes_list = [a for a in self.figure.get_axes()
-                     if a.patch.contains_point(xy)]
-
+                     if a.patch.contains_point(xy) and a.get_visible()]
```

Comment:
perhaps reverse the order of the checks?  get_visible() should be faster (just an attribute check) than contains_point()
@jklymak can self-merge with or without this change.

Reply:
The first check is to make sure the pick is actually in the axes (because its in pixels).  So if the second check failed, and you didn't have the first, you wouldn't know if it failed because of a resolution change in the tests or because the functionality actually broke.  

In normal conditions both tests should pass, so I don't think there is a typical time savings here.